src/Avalonia.DesignerSupport/DesignWindowLoader.cs

```diff
@@ -35,7 +35,16 @@ public static Window LoadDesignerWindow(string xaml, string assemblyPath, string
                 }
 
                 var localAsm = assemblyPath != null ? Assembly.LoadFile(Path.GetFullPath(assemblyPath)) : null;
-                var loaded = loader.Load(stream, localAsm, null, baseUri, true);
+                var useCompiledBindings = localAsm?.GetCustomAttributes<AssemblyMetadataAttribute>()
+                    .FirstOrDefault(a => a.Key == "AvaloniaUseCompiledBindingsByDefault")?.Value;
+
+                var loaded = loader.Load(stream, new RuntimeXamlLoaderConfiguration
+                {
+                    LocalAssembly = localAsm,
+                    BaseUri = baseUri,
+                    DesignMode = true,
+                    UseCompiledBindingsByDefault = bool.TryParse(useCompiledBindings, out var tempBool) && tempBool
```

Comment:
Nit: `tempBool` isn't descriptive enough, IMO. `parsed` or `parsedValue` would be better.